torchtext/datasets/language_modeling.py

```
@@ -1,217 +1,236 @@
+                    data_select=('train', 'test', 'valid')):
+
+    if isinstance(data_select, str):
+        data_select = [data_select]
+    for item in data_select:
+        if item not in ('train', 'test', 'valid'):
+            raise TypeError('{} in data_select is not supported!'.format(item))
+
+    if dataset_name == 'PennTreebank':
+        extracted_files = []
+        if 'train' in data_select:
+            extracted_files.append(download_from_url(URLS['PennTreebank'][0], root=root))
+        if 'test' in data_select:
+            extracted_files.append(download_from_url(URLS['PennTreebank'][1], root=root))
+        if 'valid' in data_select:
+            extracted_files.append(download_from_url(URLS['PennTreebank'][2], root=root))
+    else:
+        dataset_tar = download_from_url(URLS[dataset_name], root=root)
+        extracted_files = [os.path.join(root, d) for d in extract_archive(dataset_tar)]
+
+    _path = {}
+    for item in data_select:
+        _path[item] = _get_datafile_path(item, extracted_files)
+
+    if vocab is None:
+        if 'train' not in _path.keys():
+            raise TypeError("Must pass a vocab if train is not selected.")
+        logging.info('Building Vocab based on {}'.format(_path['train']))
+        vocab = build_vocab_from_iterator(read_text_iterator(_path['train'], tokenizer))
+        logging.info('Vocab has {} entries'.format(len(vocab)))
+    else:
+        if not isinstance(vocab, Vocab):
+            raise TypeError("Passed vocabulary is not of type Vocab")
+
+    _data = {}
+    for item in _path.keys():
+        _data[item] = []
+        logging.info('Creating {} data'.format(item))
+        _iter = create_data_from_iterator(
+            vocab, read_text_iterator(_path[item], tokenizer), removed_tokens)
+        for tokens in _iter:
+            _data[item] += tokens
+
+    return tuple(LanguageModelingDataset(torch.tensor(_data[d]).long(), vocab)
+                 for d in data_select if _data[d] != [])
```

I think if ```_data[d]``` is empty we actually want to raise an error because the data the user requested is empty OR we simply return a dataset that loads nothing.

If the data is empty we might have downloaded something corrupted.

---

Done!